internal/framework/service/dlp_datasets/data_source_test.go

```diff
@@ -0,0 +1,36 @@
+package dlp_datasets_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccCloudflareDLPDatasets_DataSource(t *testing.T) {
```

Comment:
this is currently failing with type errors

```
TF_ACC=1 go test ./internal/framework/service/dlp_datasets -v -run "^TestAccCloudflareDLPDatasets_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDLPDatasets_DataSource
=== PAUSE TestAccCloudflareDLPDatasets_DataSource
=== CONT  TestAccCloudflareDLPDatasets_DataSource
    data_source_test.go:15: Step 1/1 error: Error running pre-apply plan: exit status 1
        
        Error: Value Conversion Error
        
          with data.cloudflare_dlp_datasets.my_datasets,
          on terraform_plugin_test.tf line 12, in data "cloudflare_dlp_datasets" "my_datasets":
          12: data "cloudflare_dlp_datasets" "my_datasets" {
        
        An unexpected error was encountered while verifying an attribute value
        matched its expected type to prevent unexpected behavior or panics. This is
        always an error in the provider. Please report the following to the provider
        developer:
        
        Expected framework type from provider logic:
        types.ListType[types.ObjectType["description":basetypes.StringType,
        "id":basetypes.StringType, "name":basetypes.StringType,
        "secret":basetypes.BoolType, "status":basetypes.StringType]] / underlying
        type: tftypes.List[tftypes.Object["description":tftypes.String,
        "id":tftypes.String, "name":tftypes.String, "secret":tftypes.Bool,
        "status":tftypes.String]]
        Received framework type from provider logic:
        types.ListType[types.ObjectType[]] / underlying type:
        tftypes.List[tftypes.Object[]]
        Path: datasets
--- FAIL: TestAccCloudflareDLPDatasets_DataSource (2.79s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/dlp_datasets	3.640s
FAIL
make: *** [testacc] Error 1
```

i'm also not sure if the schema structure is correct. i think you want to define the nested types in a dedicated `Blocks` key with the nested values (like we do with the `rulesets` service. without this, i think you're essentially having a loose object representation without instead of the block notation.

Reply:
Thanks. I've made the change